src/crypto/SecretKey.cpp

```diff
@@ -327,16 +328,13 @@ PubKeyUtils::verifySig(PublicKey const& key, Signature const& signature,
 SignatureHint
 PubKeyUtils::getHint(PublicKey const& pk)
 {
-    SignatureHint res;
-    memcpy(res.data(), &pk.ed25519().back() - res.size() + 1, res.size());
-    return res;
+    return SignatureUtils::getHint(pk.ed25519());
 }
 
 bool
 PubKeyUtils::hasHint(PublicKey const& pk, SignatureHint const& hint)
```

Comment:
This seems misnamed. Seems like it should be called `doesHintMatch`  or something